tests/test_objective_funs.py

```diff
@@ -485,25 +493,284 @@ def test_boundary_error_nestor(self):
         np.testing.assert_allclose(f, 0, atol=2e-3)
 
     @pytest.mark.unit
-    def test_target_mean_iota(self):
-        """Test calculation of iota profile average."""
+    def test_quadratic_flux(self):
+        """Test calculation of quadratic flux on the boundary."""
+        t_field = ToroidalMagneticField(1, 1)
 
-        def test(eq):
-            grid = LinearGrid(L=5, M=1, N=1, NFP=eq.NFP)
-            mean_iota = jnp.mean(eq.compute("iota", grid=grid)["iota"])
-            obj = RotationalTransform(
-                target=mean_iota, weight=1, eq=eq, loss_function="mean", grid=grid
+        # test that torus (axisymmetric) Bnorm is exactly 0
+        eq = Equilibrium()
+        eq.solve()
+        obj = QuadraticFlux(t_field, eq, eq_fixed=True)
+        obj.build(eq)
+        f = obj.compute(params_1=t_field.params_dict)
+        np.testing.assert_allclose(f, 0)
+
+        # test nonaxisymmetric surface
+        eq = get("precise_QA")
+
+        source_N = 20
+        source_M = 10
+
+        eval_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=eq.M_grid,
+            N=eq.N_grid,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        source_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=source_M,
+            N=source_N,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        obj = QuadraticFlux(
+            t_field,
+            eq,
+            eval_grid=eval_grid,
+            src_grid=source_grid,
+            eq_fixed=True,
+        )
+        Bnorm = t_field.compute_Bnormal(
+            eq.surface, eval_grid=eval_grid, source_grid=source_grid
+        )[0]
+        obj.build(eq, verbose=0)
+        f = obj.compute(params_1=t_field.params_dict)
+
+        # for this to pass with rtol=1e-3, the source resolution needs to be quite high
+        np.testing.assert_allclose(f, Bnorm, rtol=1e-2, atol=1e-2)
+
+    def test_quadratic_flux_optimization(self):
+        """Test optimization using quadratic flux."""
+
+        def get_grids(eq, field):
+            N_grid = 20
+            M_grid = 10
```

Comment:
I suggest making these input parameters so they can be chosen separately for each test